src/bitmap.c

```diff
@@ -902,14 +902,14 @@ GdipCreateBitmapFromScan0 (int width, int height, int stride, PixelFormat format
 			goto fail;
 		}
 
-		if ((gdip_get_pixel_format_bpp(format) < 16) || gdip_is_an_alpha_pixelformat(format)) {
+		if ((gdip_get_pixel_format_bpp(format) < 32) || gdip_is_an_alpha_pixelformat(format)) {
```

Comment:
Looks like the code should handle *non-alpha* formats where the space for alpha channel is unused (but Cairo cares about it). Looking at the list of pixel formats the obvious ones are PixelFormat32bppRGB and PixelFormat16bppRGB555. We never really handled the 16-bit case here and it may still be broken. The `if` now basically boils down to `format != PixelFormat32bppRGB`.